Code/Framework/AzFramework/AzFramework/Visibility/VisibleGeometryBus.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Component/ComponentBus.h>
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Math/Matrix4x4.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/std/containers/vector.h>
+
+namespace AZ
+{
+    class ReflectContext;
+}
+
+namespace AzFramework
+{
+    //! VisibleGeometry describes details about visible geometry surfaces stored as generic indexed triangle lists
+    struct VisibleGeometry
+    {
+        AZ_TYPE_INFO(VisibleGeometry, "{4B011208-B105-4BC1-A4F3-FD5C44785D71}");
+        AZ_CLASS_ALLOCATOR(VisibleGeometry, AZ::SystemAllocator);
+
+        static void Reflect(AZ::ReflectContext* context);
+
+        AZ::Matrix4x4 m_transform;
+        AZStd::vector<float> m_vertices;
```

Comment:
Why not use a Vector3 or PackedVector3 here? Seems like at a minimum we should be guaranteeing that vertices have XYZ values?

Reply:
I started with Vector3 but wasn't sure if it was safe to reinterpret cast the container as a float* or memcpy. If it's safe to do that then I will switch it back to Vector3.

Reply:
PackedVector3f looks like a good candidate but it needs to be reflected. I'm not currently serializing any of the visible geometry data structures but want to maintain their accessibility from script.